python/ray/_private/runtime_env/validation.py

```diff
@@ -22,19 +23,169 @@
+                        f"dict, got {type(conda)}.")
+
+    return result
+
+
+def parse_and_validate_pip(pip: Union[str, List[str]]) -> Optional[List[str]]:
+    """Parses and validates a user-provided 'pip' option.
+
+    Conda can be one of two cases:
+        1) A List[str] describing the requirements. This is passed through.
+        2) A string pointing to a local requirements file. In this case, the
+           file contents will be read split into a list.
+    """
+    assert pip is not None
+
+    result = None
+    if sys.platform == "win32":
+        raise NotImplementedError("The 'pip' field in runtime_env "
+                                  "is not currently supported on "
+                                  "Windows.")
+    elif isinstance(pip, str):
+        # We have been given a path to a requirements.txt file.
+        pip_file = Path(pip)
+        if not pip_file.is_file():
+            raise ValueError(f"{pip_file} is not a valid file")
+        result = pip_file.read_text().strip().split("\n")
+    elif isinstance(pip, list) and all(isinstance(dep, str) for dep in pip):
+        if len(pip) == 0:
+            result = None
+        else:
+            result = pip
+    else:
+        raise TypeError("runtime_env['pip'] must be of type str or "
+                        f"List[str], got {type(pip)}")
+
+    return result
+
+
+def parse_and_validate_uris(uris: List[str]) -> List[str]:
+    """Parses and validates a user-provided 'uris' option.
+
+    These are passed through without validation (for now).
+    """
+    assert uris is not None
+    return uris
+
+
+def parse_and_validate_container(container: List[str]) -> List[str]:
+    """Parses and validates a user-provided 'container' option.
+
+    This is passed through without validation (for now).
+    """
+    assert container is not None
+    return container
+
+
+def parse_and_validate_env_vars(
+        env_vars: Dict[str, str]) -> Optional[Dict[str, str]]:
+    """Parses and validates a user-provided 'env_vars' option.
+
+    This is validated to verify that all keys and vals are strings.
+
+    If an empty dictionary is passed, we return `None` for consistency.
+    """
+    assert env_vars is not None
+    if len(env_vars) == 0:
+        return None
+
+    if not (isinstance(env_vars, dict) and all(
+            isinstance(k, str) and isinstance(v, str)
+            for (k, v) in env_vars.items())):
+        raise TypeError("runtime_env['env_vars'] must be of type "
+                        "Dict[str, str]")
+
+    return env_vars
+
+
+# Dictionary mapping runtime_env options with the function to parse and
+# validate them.
+OPTION_TO_VALIDATION_FN = {
+    "working_dir": parse_and_validate_working_dir,
+    "conda": parse_and_validate_conda,
+    "pip": parse_and_validate_pip,
+    "uris": parse_and_validate_uris,
+    "env_vars": parse_and_validate_env_vars,
+    "container": parse_and_validate_container,
+}
 
-    Attributes:
+
+class ParsedRuntimeEnv(dict):
```

Comment:
not relevant to this PR, but do we have any type checking / schema for runtime env ? Looks like it's just a dictionary that we've been throwing around and merging fields ..

Reply:
Yeah, you're unfortunately right haha. This would be a good thing to also add schema validation to when we do it for job submission :)